File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    var answer = 0;
    for (let i = left; i <= right; i++) {
        if (Number.isInteger(Math.sqrt(i))) { // 제곱근이 정수면 약수의 개수는 홀수
            answer -= i;
        } else {
            answer += i;
        }
    }
    return answer;
}

// https://velog.io/@leeeunbin/%ED%94%84%EB%A1%9C%EA%B7%B8%EB%9E%98%EB%A8%B8%EC%8A%A4-%EC%95%BD%EC%88%98%EC%9D%98-%EA%B0%9C%EC%88%98%EC%99%80-%EB%8D%A7%EC%85%88-JavaScript

File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0    
    
    for i in range(left, right + 1):
        measure_list = []
        for j in range(1, i + 1):
            if i % j == 0:
                measure_list.append(j)
        if len(measure_list) % 2 == 0:
            answer += i
        else:
            answer -= i    
            
    return answer